src/main/java/roomescape/service/payment/PaymentRestClient.java

```diff
@@ -0,0 +1,94 @@
+package roomescape.service.payment;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClientResponseException;
+import roomescape.domain.payment.Payment;
+import roomescape.exception.PaymentException;
+import roomescape.service.payment.dto.PaymentErrorResult;
+import roomescape.service.payment.dto.PaymentResult;
+import roomescape.service.reservation.dto.ReservationRequest;
+
+public class PaymentRestClient {
+
+    private static final String CONFIRM_ENDPOINT = "/confirm";
+    private static final String CANCEL_ENDPOINT = "/cancel";
+
+    @Value("${tosspay.secret_key}")
+    private String secretKey;
+
+    private final RestClient restClient;
+    private final ObjectMapper objectMapper;
+
+    public PaymentRestClient(RestClient restClient, ObjectMapper objectMapper) {
+        this.restClient = restClient;
+        this.objectMapper = objectMapper;
+    }
+
+    public PaymentResult confirm(ReservationRequest request) {
+        HttpHeaders headers = generateHttpHeaders();
+        try {
+            return restClient.post()
+                    .uri(CONFIRM_ENDPOINT)
+                    .headers(httpHeaders -> httpHeaders.addAll(headers))
+                    .body(generatedConfirmRequestBody(request))
+                    .retrieve()
+                    .body(PaymentResult.class);
+        } catch (RestClientResponseException exception) {
+            String responseBody = exception.getResponseBodyAsString();
+            throw new PaymentException(parseErrorBody(responseBody));
+        }
+    }
+
+    private Map<String, Object> generatedConfirmRequestBody(ReservationRequest request) {
+        return Map.of(
+                "amount", request.amount(),
+                "orderId", request.orderId(),
+                "paymentKey", request.paymentKey()
+        );
+    }
+
+    public void cancel(Payment payment) {
+        HttpHeaders headers = generateHttpHeaders();
+        try {
+            restClient.post()
+                    .uri("/" + payment.getPaymentKey() + CANCEL_ENDPOINT)
```

Comment:
```suggestion
                    .uri(String.format("/%s/cancel", payment.getPaymentKey()))
```
uri 는 보통 이렇게 처리하는게 가독성이 좋습니다!

Reply:
오.. 저 부분이 좀 마음에 들지 않았었는데 훨씬 깔끔한 것 같아요 감사합니다.